src/document_data.js

```diff
@@ -0,0 +1,251 @@
+import { cleanUpLine, Line, updateLine } from "./line/line_data";
```

Comment:
I'd be tempted to move the chunk prototypes to `Doc.js` and the update logic to `changes.js`, and drop this module (since it mostly contains stuff that I wouldn't expect here, based on the filename)